Algorithms/Divide-and-Conquer/heap.c

```diff
@@ -0,0 +1,81 @@
+#include <stdio.h>
+#include<malloc.h>
+/*Name - Jyoti Mishra (jstjyoti)
+Email-jyotimishra29091997@gmail.com
+Heap-Sort (Divide and Conquer) 
+INTRODUCTION :  
+a. Definition of a complete Binary Tree: A  Tree is said to be complete binary tree if all it's levels have maximum number of nodes(two for each parent) except possibly the last level and 
+all the nodes in the last level appear as per left as possible. 
+b. Heap: A Heap is a Complete Binary Tree which is implemented using an array(Linear Data Structure). 
+Max Heap: A heap is called max heap if all the nodes having a value greater than every children of the node. 
+This is used for Sorting in Ascending Order. 
+Min Heap: a heap is called min heap if all the nodes having smaller value than every children of the node '''
+*/
+void swap(int *x,int *y)
+{
+	int t=*x;
+	*x=*y;
+	*y=t;
+}
+/*This function is for swapping two values having Time Complexity O(1) and space Complexity O(1) as only one estra pointer required to swap values*/

```

Comment:
Fix typo ~estra~ to extra. Please avoid making typing errors in comments so that comments can be read smoothly. And always write comments above the function, not below it.